lua/acf/scanner/scanner_sh.lua

```diff
@@ -0,0 +1,1472 @@
+        render.SetMaterial(wireframe)
+
+        physMeshCache[ent].mesh:Draw()
+
+        render.SetStencilCompareFunction(STENCIL_EQUAL)
+        render.ClearBuffersObeyStencil(color.r, color.g, color.b, color.a, true);
+
+        render.SetStencilEnable(false)
+
+        cam.PopModelMatrix()
+    end
+
+    local function drawBounds(ent, scanDef)
+        if not IsValid(ent) then return end
+        local mi, ma
+        if not IsValid(ent:GetPhysicsObject()) then
+            if ent2bp[ent] then
+                mi = ent2bp[ent][1]
+                ma = ent2bp[ent][2]
+            end
+        else
+            mi, ma = ent:GetPhysicsObject():GetAABB()
+        end
+
+        if mi == nil or ma == nil then return end
+
+        render.SetColorMaterial()
+        render.DepthRange(0, 0.01)
+            render.DrawBox(ent:GetPos(), ent:GetAngles(), mi, ma, scanDef.colorBoundsInside)
+            render.DrawWireframeBox(ent:GetPos(), ent:GetAngles(), mi, ma, scanDef.colorBounds)
+        render.DepthRange(0, 1)
+    end
+
+    --local eyetrace_calc = nil
+    local player_mt = FindMetaTable("Player")
+    if not ACF_SCANNING_PLMT_GETEYETRACE then
+        ACF_SCANNING_PLMT_GETEYETRACE = player_mt.GetEyeTrace
+    end
+
+    local plmt_eyetrace = ACF_SCANNING_PLMT_GETEYETRACE
+    local lastFrame, lastFrameTrace
+
+    function player_mt:GetEyeTrace()
+        if not scanning.IsScannerActive() then
+            return plmt_eyetrace(self)
+        end
+        local frameNum = FrameNumber()
+        if lastFrame == frameNum then
+            return lastFrameTrace
+        end
+
+        lastFrame = frameNum
+        local tr = util.TraceLine{
+            start = calcview_lastPos,
+            endpos = calcview_lastPos + (calcview_lastAng:Forward() * 32768),
+            filter = self
+        }
+        lastFrameTrace = tr
+        return tr
+    end
+
+    local function VisualizeClips(ent)
+        if not IsValid(ent) then return end
+        local clips = getClips(ent)
+        local clipL = #clips
+        if clipL == 0 then return end
+
+        local clipR = 30
+        if clipL > 0 then
+            for _, clipPlane in ipairs(clips) do
+                render.DepthRange(0, 0.01)
+                local localEnt = clipPlane.local_ent
+                local isLocalEntValid = IsValid(localEnt)
+                local origin = isLocalEntValid and localEnt:LocalToWorld(clipPlane.origin) or clipPlane.origin
+                local normal = (isLocalEntValid and LocalToWorldVector(localEnt, clipPlane.normal) or clipPlane.normal)
+                local angles = normal:Angle()
+                render.DrawBox(
+                    origin,
+                    angles,
+                    Vector(-0.02, -clipR, -clipR),
+                    Vector(0.02, clipR, clipR),
+                    Color(50, 255, 50, 70)
+                    )
+
+                local c1 = LocalToWorld(Vector(0, -clipR, -clipR), Angle(0, 0, 0), origin, angles)
```

Comment:
Please use `angle_zero` for these angles and localize the colors used below for optimization purposes